src/init/AppRunner.ts

```diff
@@ -57,96 +50,80 @@ export class AppRunner {
+    const loaderProperties = {
+      mainModulePath: resolveAssetPath(params.mainModulePath as string),
       dumpErrorState: true,
       logLevel: params.loggingLevel as LogLevel,
     };
-    const configFile = resolveAssetPath(params.config);
 
-    // Create and execute the app
-    this.createApp(loaderProperties, configFile, params)
-      .then(
-        async(app): Promise<void> => app.start(),
-        (error: Error): void => {
-          // Instantiation of components has failed, so there is no logger to use
-          stderr.write(`Error: could not instantiate server from ${configFile}\n`);
-          stderr.write(`${error.stack}\n`);
-          process.exit(1);
-        },
-      ).catch((error): void => {
-        this.logger.error(`Could not start server: ${error}`, { error });
-        process.exit(1);
-      });
+    // Create varResolver
+    const varConfigFile = resolveAssetPath(params.varConfig as string);
+    this.createComponent(
+      loaderProperties as IComponentsManagerBuilderOptions<VarResolver>, varConfigFile, {}, varConfigComponentIri,
+    ).then(
+    //  Using varResolver resolve vars and start app
+      async(varResolver): Promise<void> => {
+        let vars;
+        try {
+          vars = await varResolver.resolveVars(argv);
+        } catch (error: unknown) {
+          this.exitWithError(error as Error, 'Error in computing variables', stderr);
+        }
+        return this.initApp(
+          loaderProperties, resolveAssetPath(params.config as string), vars as unknown as Record<string, any>, stderr,
+        );
+      },
+
+      (error): void => this.exitWithError(
+        error, `Error in loading variable configuration from ${varConfigFile}\n`, stderr,
+      ),
+    ).catch((error): void => this.exitWithError(error, 'Could not start the server', stderr));
   }
 
-  /**
-   * Creates the main app object to start the server from a given config.
-   * @param loaderProperties - Components.js loader properties.
-   * @param configFile - Path to a Components.js config file.
-   * @param variables - Variables to pass into the config file.
-   */
-  public async createApp(
+  private exitWithError(error: Error, message: string, stderr: WriteStream): void {
+    stderr.write(message);
+    stderr.write(`${error.stack}\n`);
+    process.exit(1);
+  }
+
+  private async initApp(
     loaderProperties: IComponentsManagerBuilderOptions<App>,
-    configFile: string,
-    variables: CliParams | Record<string, any>,
-  ): Promise<App> {
-    // Translate command-line parameters if needed
-    if (typeof variables.loggingLevel === 'string') {
-      variables = this.createVariables(variables as CliParams);
+    configFile: string, vars: Record<string, any>, stderr: WriteStream,
+  ): Promise<void> {
+    let app;
+    // Create app
+    try {
+      app = await this.createComponent(
+        loaderProperties, configFile, vars, appComponentIri,
+      );
```

Comment:
```suggestion
      app = await this.createComponent(loaderProperties, configFile, vars, appComponentIri);
```
Preferably try to not put parameters on new lines as long as the linter doesn't complain.